app/models/question_type.rb

```diff
@@ -3,16 +3,22 @@ class QuestionType
   attr_reader :name, :odk_name, :properties
 
   @@attributes = [
-    {:name => "text", :odk_name => "string", :properties => %w(form_printable smsable textual headerable)},
-    {:name => "long_text", :odk_name => "string", :properties => %w(form_printable smsable textual)},
-    {:name => "integer", :odk_name => "int", :properties => %w(form_printable smsable numeric headerable)},
-    {:name => "decimal", :odk_name => "decimal", :properties => %w(form_printable smsable numeric headerable)},
-    {:name => "location", :odk_name => "geopoint", :properties => %w()},
-    {:name => "select_one", :odk_name => "select1", :properties => %w(form_printable has_options smsable headerable)},
-    {:name => "select_multiple", :odk_name => "select", :properties => %w(form_printable has_options smsable headerable)},
-    {:name => "datetime", :odk_name => "dateTime", :properties => %w(form_printable temporal has_timezone smsable headerable)},
-    {:name => "date", :odk_name => "date", :properties => %w(form_printable temporal smsable headerable)},
-    {:name => "time", :odk_name => "time", :properties => %w(form_printable temporal smsable headerable)}
+    { name: 'text', odk_name: 'string', properties: %w(form_printable smsable textual headerable) },
+    { name: 'long_text', odk_name: 'string', properties: %w(form_printable smsable textual) },
+    { name: 'integer', odk_name: 'int', properties: %w(form_printable smsable numeric headerable) },
+    { name: 'decimal', odk_name: 'decimal', properties: %w(form_printable smsable numeric headerable) },
+    { name: 'location', odk_name: 'geopoint', properties: %w() },
+    { name: 'select_one', odk_name: 'select1', properties: %w(form_printable has_options smsable headerable) },
+    { name: 'select_multiple', odk_name: 'select', properties: %w(form_printable has_options smsable headerable) },
+    { name: 'datetime', odk_name: 'dateTime', properties: %w(form_printable temporal has_timezone smsable headerable) },
+    { name: 'date', odk_name: 'date', properties: %w(form_printable temporal smsable headerable) },
+    { name: 'time', odk_name: 'time', properties: %w(form_printable temporal smsable headerable) },
+    { name: 'image', odk_name: 'image/*', properties: %w(multimedia) },
```

Comment:
Right. That would actually perhaps be a good time to strip odk_name out of
here.

On Thu, Jan 21, 2016 at 2:55 PM, Melody <notifications@github.com> wrote:

> In app/models/question_type.rb
> <https://github.com/thecartercenter/elmo/pull/175#discussion_r50453932>:
>
> > -    {:name => "select_one", :odk_name => "select1", :properties => %w(form_printable has_options smsable headerable)},
> > -    {:name => "select_multiple", :odk_name => "select", :properties => %w(form_printable has_options smsable headerable)},
> > -    {:name => "datetime", :odk_name => "dateTime", :properties => %w(form_printable temporal has_timezone smsable headerable)},
> > -    {:name => "date", :odk_name => "date", :properties => %w(form_printable temporal smsable headerable)},
> > -    {:name => "time", :odk_name => "time", :properties => %w(form_printable temporal smsable headerable)}
> > +    { name: 'text', odk_name: 'string', properties: %w(form_printable smsable textual headerable) },
> > +    { name: 'long_text', odk_name: 'string', properties: %w(form_printable smsable textual) },
> > +    { name: 'integer', odk_name: 'int', properties: %w(form_printable smsable numeric headerable) },
> > +    { name: 'decimal', odk_name: 'decimal', properties: %w(form_printable smsable numeric headerable) },
> > +    { name: 'location', odk_name: 'geopoint', properties: %w() },
> > +    { name: 'select_one', odk_name: 'select1', properties: %w(form_printable has_options smsable headerable) },
> > +    { name: 'select_multiple', odk_name: 'select', properties: %w(form_printable has_options smsable headerable) },
> > +    { name: 'datetime', odk_name: 'dateTime', properties: %w(form_printable temporal has_timezone smsable headerable) },
> > +    { name: 'date', odk_name: 'date', properties: %w(form_printable temporal smsable headerable) },
> > +    { name: 'time', odk_name: 'time', properties: %w(form_printable temporal smsable headerable) },
> > +    { name: 'image', odk_name: 'image/*', properties: %w(multimedia) },
>
> Gotcha, so odk_name refers specifically to the part that goes into the
> bind and we'll need to incorporate the mediatype thing separately when we
> deal with the rendering portion?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thecartercenter/elmo/pull/175/files#r50453932>.
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org